lib/trace-agent.js

```diff
@@ -163,8 +163,11 @@ TraceAgent.prototype.addTransactionLabel = function(key, value) {
  * Determines whether a trace of the given name should be recorded based
  * on the current tracing policy.
  */
-TraceAgent.prototype.shouldTrace = function(name) {
-  return this.policy.shouldTrace(Date.now(), name);
+TraceAgent.prototype.shouldTrace = function(name, options) {
+  if (isNaN(options)) {
```

Comment:
Okay, so reading this my understanding of the semantics of whether we should trace or not are:

* If there is no trace context header; we use our sampling
* If there is a trace context header, but no options; we use our sampling
* If there is a trace context header with options, and options is a number; we use that to decide.

Can you edit the PR abstract to include the rationale (why we need to change this) and summarize the new semantics?